### PR TITLE
docs: add docs for user routes

### PR DESCRIPTION
**Pull request by Sukhacoder02**
- added swagger auto-generated docs for all user routes - GET, POST, PUT, DELETE.
- updated repository's README's setup steps to include cloning the repo with Https also.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,16 +2,26 @@
...
+
 ```
 git clone git@github.com:tech-university-india/staffing-pipeline-backend.git
 ```
 
+**Clone Repository with Https**
+
+```
+git clone https://github.com/tech-university-india/staffing-pipeline-backend.git
+```
```
Comment (rohini-senthil) [inline README.md:15]: I don't think this is necessary to add in

Comment (Sukhacoder02) [reply README.md]: Removed, Waiting for review.

**Comment by rohini-senthil** [comment]
> 'accessLogStream' is assigned a value but never used

I there a reason why the following line is present in index.js:
```
const accessLogStream = fs.createWriteStream(path.join(__dirname + '/../', 'access.log'), { flags: 'a' });
```

File: swagger.json [hunk 2]
```diff
@@ -5,54 +5,172 @@
...
+  "host": "localhost:3000",
   "basePath": "/",
-  "schemes": ["http"],
+  "schemes": [
+    "http"
+  ],
   "paths": {
-    "/login": {
+    "/users": {
+      "get": {
```
Comment (rohini-senthil) [inline swagger.json:15]: Let's add examples for response & request objects? 

https://swagger.io/docs/specification/adding-examples/

File: swagger.json [hunk 3]
```diff
@@ -5,54 +5,172 @@
     "title": "REST API",
     "description": ""
   },
-  "host": "localhost:8000",
```
Comment (rohini-senthil) [inline swagger.json:8]: The port defined in index.js is 8000

File: swagger.json [hunk 4]
```diff
@@ -5,54 +5,172 @@
...
       "get": {
-        "description": "",
-        "parameters": [],
+        "description": "retrieves the user with given 'userId' from the database",
+        "parameters": [
+          {
+            "name": "userId",
+            "in": "path",
+            "required": true,
+            "type": "string"
```
Comment (rohini-senthil) [inline swagger.json:92]: Is it possible to mention it is UUID here?

Comment (Sukhacoder02) [reply swagger.json:92]: > > 'accessLogStream' is assigned a value but never used
> 
> I there a reason why the following line is present in index.js:
> 
> ```
> const accessLogStream = fs.createWriteStream(path.join(__dirname + '/../', 'access.log'), { flags: 'a' });
> ```

It is actually present in current main before I made changes in my branch, therefore I didn't change/remove it in-case it was being used somewhere else, but it doesn't look like it is doing something.
To be honest, I also have not idea why its there 😅.

Comment (Sukhacoder02) [reply swagger.json:92]: Also, it breaks the code. So, should I remove it?

Comment (rohini-senthil) [reply swagger.json:92]: Yep, let's remove it please

**Comment by Sukhacoder02** [comment]
Made the changes as directed, Rohini.

**Review by rohini-senthil — APPROVED** [review]

**Pull request merged**